### ISSUE TITLE
LWG4113: Disallow `has_unique_object_representations<Incomplete[]>`

### ISSUE THREAD

**Issue by ldionne** [llvm/llvm-project#118350]
**Link:** https://wg21.link/LWG4113

**Comment by frederick-vs-ja** [comment]
It seems better to let compiler frontends to handle this.

**Comment by philnik777** [comment]
> It seems better to let compiler frontends to handle this.

True, but we should add tests for it still. We're actually lacking any kind of test coverage to ensure we're rejecting incorrect type traits uses properly.

**Comment by cor3ntin** [comment]
@philnik777 I modified the builtin, I don't know if you want to add more tests on the libc++ side

**Comment by philnik777** [comment]
@cor3ntin yeah, I think we should keep this open until we've added tests on the libc++ side.

**Comment by ldionne** [comment]
@philnik777 I'll close this since we don't want to have a conformance issue marked as done but still open. Instead I'll create another issue to add tests for this.

**Comment by philnik777** [comment]
@ldionne I don't think we should claim conformance if we don't test for it, which IMO we effectively do by closing this.

**Comment by ldionne** [comment]
Hmm, in that case what we should have done is avoid marking the issue as "complete" in the `libc++ Standards Conformance` project. Instead of shuffling everything around, I think we should just add the tests. I discovered this while synchronizing the status pages with the Github issues.

**Comment by philnik777** [comment]
@ldionne I'm not sure I understand. Where was it marked as complete?

**Comment by ldionne** [comment]
When it was originally closed via the Clang PR, it was marked as done in the GH project. Then it was reopened, which left it in the state where the Issue was opened but it was marked as Done.

**Comment by philnik777** [comment]
That seems problematic. What if a PR marks an issue as complete but is reverted for some reason? I don't think we want something ever marked as done if the related issue is still open. How is the status on the page updated? Can we influence that somehow?

**Comment by ldionne** [comment]
> That seems problematic. What if a PR marks an issue as complete but is reverted for some reason?

Then I think we should move the Github issue back to `Todo` in the Github project page.

> I don't think we want something ever marked as done if the related issue is still open. How is the status on the page updated? Can we influence that somehow?

The `.csv` status files are updated manually when people make PRs, and I synchronize them with `libcxx/utils/synchronize_csv_status_files.py` from time to time (this mostly catches oversights).

**Comment by ldionne** [comment]
I don't think this is a huge problem. In this case, the Github project entry should have been moved back to Todo when the issue was reopened, but the discrepancy was flagged by the script as intended, so nothing bad happened.
